src/pyuvsim/simsetup.py

```diff
@@ -1911,129 +2012,109 @@ def initialize_uvdata_from_params(
+            cat_name = "unprojected"
     elif "object_name" in param_dict:
-        uvparam_dict[uvparam_name] = param_dict["object_name"]
+        cat_name = param_dict["object_name"]
     else:
-        uvparam_dict[uvparam_name] = param_dict["cat_name"]
-
-    uv_obj = UVData()
-    uv_obj._set_future_array_shapes()
-
-    # Setting the frame and acceptable range for the moon
-    uv_obj._telescope_location.frame = uvparam_dict["telescope_frame"]
-    uv_obj._telescope_location.ellipsoid = uvparam_dict["ellipsoid"]
-
-    # use the __iter__ function on UVData to get list of UVParameters on UVData
-    valid_param_names = [getattr(uv_obj, param).name for param in uv_obj]
-    for k in valid_param_names:
-        if k in param_dict:
-            setattr(uv_obj, k, param_dict[k])
-        if k in uvparam_dict:
-            setattr(uv_obj, k, uvparam_dict[k])
-    logger.info("Initialized UVData object")
-
-    bls = np.array(
-        [
-            uv_obj.antnums_to_baseline(
-                uv_obj.antenna_numbers[j], uv_obj.antenna_numbers[i]
-            )
-            for i in range(0, uv_obj.Nants_data)
-            for j in range(i, uv_obj.Nants_data)
-        ]
-    )
-
-    uv_obj.baseline_array = np.tile(bls, uv_obj.Ntimes)
-    uv_obj.Nbls = bls.size
-    uv_obj.time_array = np.repeat(uv_obj.time_array, uv_obj.Nbls)
-    uv_obj.integration_time = np.repeat(
-        uv_obj.integration_time, uv_obj.Nbls * uv_obj.Ntimes
-    )
-    uv_obj.Nblts = uv_obj.Nbls * uv_obj.Ntimes
-    uv_obj.blt_order = ("time", "ant1")
-    uv_obj.ant_1_array, uv_obj.ant_2_array = uv_obj.baseline_to_antnums(
-        uv_obj.baseline_array
-    )
-
-    logger.info(f"BLT-ORDER: {uv_obj.blt_order}")
-    cat_id = uv_obj._add_phase_center(
-        cat_name=uvparam_dict["cat_name"], cat_type="unprojected"
-    )
-    uv_obj.phase_center_id_array = np.full(uv_obj.Nblts, cat_id, dtype=int)
+        cat_name = param_dict["cat_name"]
+    phase_center_catalog = {0: {"cat_name": cat_name, "cat_type": "unprojected"}}
+
+    # remove the pragma below after pyuvdata v3.0 is released
+    if hasattr(UVData(), "telescope"):  # pragma: nocover
+        tel_init_params = {"location": telescope_location}
+
+        telescope_param_map = {
+            "telescope_name": "name",
+            "antenna_names": "antenna_names",
+            "antenna_numbers": "antenna_numbers",
+            "antenna_positions": "antenna_positions",
+            "antenna_diameters": "antenna_diameters",
+            "x_orientation": "x_orientation",
+            "instrument": "instrument",
+        }
+        for key, tel_key in telescope_param_map.items():
+            if key in param_dict:
+                tel_init_params[tel_key] = param_dict[key]
+            if key in tele_params:
+                tel_init_params[tel_key] = tele_params[key]
+        if "instrument" not in tel_init_params:
+            tel_init_params["instrument"] = tel_init_params["name"]
```

Comment:
I would actually try to move to a more native configuration format that collected the telescope parameters under a heading properly, like

```yaml
telescope:
    name: hera
    antenna_names: ...
    etc.
```

It feels natural to me to be able to do `tele = Telescope.from_yaml(yamlfile)`, where the yaml file is self-contained.

Reply:
I can look at that, but I was trying not to make changes to the pyuvsim API.

Reply:
Yeah, that's a fair concern, of course. Probably both could be supported for some time. Maybe this is something to bring up in a separate issue.